src/fipp/engine.cljc

```diff
@@ -229,10 +235,9 @@
 (defn pprint-document [document options]
   (let [options (merge {:width 70} options)]
     (->> (serialize document)
-         (eduction
-           annotate-rights
-           (annotate-begins options)
-           (format-nodes options))
+         (sequence (comp annotate-rights
```

Comment:
There is an overhead to creating a chunked sequence when using `sequence`, yes, and naively I thought this would be faster:

```clojure
(->> (serialize document)
     (transduce (comp annotate-rights
                      (annotate-begins options)
                      (format-nodes options))
                (completing #(print %2))
                nil))
```

This is essentially what `(run! the-eduction print)` was doing, without the indirection of an eduction.

But in my testing, the `sequence` version is ~10% faster (using the existing benchmark and a silly microbenchmark printing `(range 100000)`). I _think_ this is because the [`IChunkedSeq`](https://github.com/clojure/clojure/blob/bc186508ab98514780efbbddb002bf6fd2938aee/src/clj/clojure/core/protocols.clj#L131) path of `InternalReduce` is being used rather than the [default](https://github.com/clojure/clojure/blob/bc186508ab98514780efbbddb002bf6fd2938aee/src/clj/clojure/core/protocols.clj#L158). I presume the chunking optimisation is winning over the overhead of having to create the chunked sequence.